MayssaraOmar/assignment1/Problem 1.cpp

```diff
@@ -18,54 +18,71 @@ vector<string> split_spaces(string sentence){
     int last = 0;
      while(last<sentence.length()){
         int next = sentence.find(' ', last);
-        string word = sentence.substr(last, next-last);
-        if(word != "")
+        string word = sentence.substr(last, next-last); // when next is npos (-1), next-last is a very big number because the second parameter of substr is unsigned
+        if(!word.empty()){
             tokens.push_back(word);
-        if(next ==  string::npos)
+        }
+        if(next ==  string::npos){
             break;
+        }
         last = next+1;
     }
 
     return tokens;
 
 }
 
-map< vector<int>,int > get_map(string sentence1){
-
+map< vector<int>,int > get_map(string sentence1, bool *error){
```

Comment:
Usually, it's easier to return bool for success and use the out parameter for the map in this case. That way you can call the function on a if statement to handle the error.